### PR TITLE
Updated dependencies  (fixes compilation error with edge `rustc`)

### PR DESCRIPTION
**Pull request by ink-splatters**
## What

- `cargo update` and fixing consequences

### TL;DR 

this may seem opinionated however I believe it's justified by compilation error on edge `rustc` which requires updating `ahash` and is hard to fix without bulk update given the current dependency tree.

### Long version

I got this compilation [error \[E0635\]: unknown feature `stdsimd`](https://users.rust-lang.org/t/error-e0635-unknown-feature-stdsimd/106445/2)

using the latest `rustc`:

```sh
$ date
Fri May 31 11:45:37 CEST 2024
$ rustc --version
rustc 1.80.0-nightly (6f3df08aa 2024-05-30)
```

In particular, it breaks [ahash](https://users.rust-lang.org/t/error-e0635-unknown-feature-stdsimd/106445/3) before [`0.8.7`](https://github.com/tkaitchuck/aHash/issues/200) given the latest `rustc` is used.

#### Two times `ahash`

in the current HEAD, `ahash` is linked 2 times by `cursive`. While it could've been possible to update `cursive` alone, it seemed to be quite brittle, hence `cargo update` was preferred.

<details>
<summary>
Click for details
</summary>

```sh
$ git checkout 847b9c7
HEAD is now at 847b9c7 Merge pull request #24 from szabgab/patch-1
$
$ rg ahash
Cargo.lock
21:name = "ahash"
32:name = "ahash"
197: "ahash 0.8.3",
218: "ahash 0.7.6",
$
$ cargo tree | rg cursive -A 10
airstation:launchk ic$ cargo tree | rg cursive -A 10
├── cursive v0.20.0
│   ├── ahash v0.8.3
│   │   ├── cfg-if v1.0.0
│   │   ├── getrandom v0.2.7
│   │   │   ├── cfg-if v1.0.0
│   │   │   └── libc v0.2.147
│   │   └── once_cell v1.14.0
│   │   [build-dependencies]
│   │   └── version_check v0.9.4
│   ├── cfg-if v1.0.0
│   ├── crossbeam-channel v0.5.6
--
│   ├── cursive_core v0.3.4
│   │   ├── ahash v0.7.6
│   │   │   ├── getrandom v0.2.7 (*)
│   │   │   └── once_cell v1.14.0
│   │   │   [build-dependencies]
│   │   │   └── version_check v0.9.4
│   │   ├── crossbeam-channel v0.5.6 (*)
│   │   ├── enum-map v2.4.1
│   │   │   └── enum-map-derive v0.10.0 (proc-macro)
│   │   │       ├── proc-macro2 v1.0.66 (*)
│   │   │       ├── quote v1.0.32 (*)

$ cargo tree  | rg cursive -A 10 | rg ahash
│   ├── ahash v0.8.3
│   │   ├── ahash v0.7.6
```

</details>

#### `new_debouncer()` fix

the update required fixing `new_debouncer()` call as the interface has slightly changed

## Tests

```sh
$ cargo install cargo-pretty-test
$ cargo-pretty-test
...
```

<details>
<summary>Click for test results</summary>

```
Generated by cargo-pretty-test
└── (OK) xpc_sys ... (21 tests in 0.00s: ✅ 21)
    └── (OK) src/lib.rs ... (21 tests in 0.00s: ✅ 21)
        ├─ ✅ bindgen_test_layout_ipc_info_name
        ├─ ✅ bindgen_test_layout_ipc_info_space
        ├─ ✅ bindgen_test_layout_ipc_info_space_basic
        ├─ ✅ bindgen_test_layout_ipc_info_tree_name
        ├─ ✅ bindgen_test_layout_mach_port_limits
        ├─ ✅ bindgen_test_layout_mach_port_options
        ├─ ✅ bindgen_test_layout_mach_port_options__bindgen_ty_1
        ├─ ✅ bindgen_test_layout_mach_port_qos
        ├─ ✅ bindgen_test_layout_mach_service_port_info
        ├── objects
        │   ├── xpc_dictionary
        │   │   └── tests
        │   │       ├─ ✅ hashmap_to_raw
        │   │       └─ ✅ raw_to_hashmap
        │   └── xpc_object
        │       └── tests
        │           └─ ✅ safely_get_refs
        ├── tests
        │   └─ ✅ it_works
        └── traits
            └── xpc_value
                └── tests
                    ├─ ✅ xpc_to_rs_with_wrong_type
                    ├─ ✅ xpc_value_array
                    ├─ ✅ xpc_value_bool
                    ├─ ✅ xpc_value_f64
                    ├─ ✅ xpc_value_i64
                    ├─ ✅ xpc_value_mach_recv
                    ├─ ✅ xpc_value_mach_send
                    └─ ✅ xpc_value_u64

Status: OK; total 21 tests in 0.00s: 21 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out

```

</details>

### CODE REVIEW

File: launchk/src/launchd/plist.rs [hunk 1]
```diff
@@ -104,7 +104,7 @@ pub const SYSTEM_LAUNCH_DAEMONS: &str = "/System/Library/LaunchDaemons";
 
 async fn fsnotify_subscriber() {
     let (tx, rx): (Sender<DebounceEventResult>, Receiver<DebounceEventResult>) = channel();
-    let mut debouncer = new_debouncer(Duration::from_secs(5), None, tx).unwrap();
+    let mut debouncer = new_debouncer(Duration::from_secs(5), tx).unwrap();
```
Comment (mach-kernel) [inline launchk/src/launchd/plist.rs:107]: Makes sense, [for reference](https://docs.rs/notify-debouncer-mini/latest/notify_debouncer_mini/fn.new_debouncer.html)

**Review by mach-kernel — APPROVED** [review]
Hi @ink-splatters!

Thank you very much for your PR! This looks great. Merging!
